### PR TITLE
Fixes #36843 - run katello-certs-check in fp-certs-generate

### PR DESCRIPTION
**Pull request by ekohl**
Untested now, but submitting for Packit.

### CODE REVIEW

**Comment by ekohl** [comment]
This came up while reviewing https://github.com/theforeman/foreman-documentation/pull/2522 and I think the whole workflow isn't great for users. The aim is run fewer commands (as a user).

**Comment by ekohl** [comment]
Updated. Now locally verified it works in both cases where there are and there aren't custom certs. Also added some other improvements to katello-certs-check.

File: bin/foreman-proxy-certs-generate [hunk 1]
```diff
@@ -28,5 +28,32 @@ Kafo::KafoConfigure.hooking.register_pre(:init) do
   org_param.value = organization
 end
 
+Kafo::KafoConfigure.hooking.register_pre_commit(:certs_check) do
+  ca_file   = param('certs', 'server_ca_cert').value
+  cert_file = param('certs', 'server_cert').value
+  key_file  = param('certs', 'server_key').value
+
+  unless app_value(:certs_skip_check) || [ca_file, cert_file, key_file].all?(&:nil?)
+    command = ['katello-certs-check', '-t', 'foreman-proxy', '-c', cert_file, '-k', key_file, '-b', ca_file]
```
Comment (ekohl) [inline bin/foreman-proxy-certs-generate]: This uses the unbranded version. Now I'm debating if we should even pass it at all. Right now it's only used to output commands, which are discarded if the command passes. Should we drop `-t foreman-proxy`?

Comment (ehelms) [reply bin/foreman-proxy-certs-generate]: Pass what?

Comment (ekohl) [reply bin/foreman-proxy-certs-generate]: The `-t foreman-proxy` part. For Satellite it should be `-t capsule`, but looking at the code this is the only place it's used:
https://github.com/theforeman/foreman-installer/blob/8b7b58dfd4b5d50a391b6c21d9b8db777896fc7f/bin/katello-certs-check#L251C7-L251C7

It's only printed if the exit code is 0, but then we discard the output. So should it be
```suggestion
    command = ['katello-certs-check', '-c', cert_file, '-k', key_file, '-b', ca_file]
```

Or do we imagine the target to become something we care about in our checks?

Comment (ehelms) [reply bin/foreman-proxy-certs-generate]: Yes - let's drop it.

**Review by evgeni — APPROVED** [review]

**Pull request merged**